ui/src/shared/utils/convertUserInput.ts

```diff
@@ -0,0 +1,5 @@
+import React from 'react'
+
+export const convertUserInputToNumOrNaN = (
+  e: React.ChangeEvent<HTMLInputElement>
+) => (e.target.value === '' ? NaN : Number(e.target.value))
```

Comment:
🤔 this ternary might cause us issues if `value` is `undefined`. Is there a possibility that the event won't have a `target` or `value` property?

Reply:
I think React.ChangeEvents always have a target. As long as we have target, value can be undefined. When value is undefined, the ternary will evaluate to false and Number(undefined) will be NaN.

Reply:
gotcha. Sweet!